components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/listener/IdentityUserIdResolverListener.java

```diff
@@ -531,7 +531,16 @@ public boolean doPreAddRole(String roleName, String[] userList, Permission[] per
             return true;
         }
 
-        String[] userIDs = getUserIdsFromUserNames(userList, (AbstractUserStoreManager) userStoreManager);
+        String[] userIDs;
+        if (ArrayUtils.isNotEmpty(userList)) {
+            userIDs = getUserIdsFromUserNames(userList, (AbstractUserStoreManager) userStoreManager);
+
+            if (ArrayUtils.isEmpty(userIDs)) {
+                return true;
+            }
+        } else {
+            userIDs = new String[0];
+        }
```

Comment:
Shall we use a turnery operator here to improve readability and reduce empty check on the userIDs?

`userIDs = ArrayUtils.isNotEmpty(userList) ? getUserIdsFromUserNames(userList, (AbstractUserStoreManager) userStoreManager) : new String[0];`

Reply:
We need to return only if the `userList` is not empty and `userIDs` array returned from the `getUserIdsFromUserNames` method invocation is empty. So the above mentioned approach will not work.